### DISCUSSION TITLE
Update TVN and TVMax from Panama

### DISCUSSION THREAD

**Discussion by KobayashiSan507** [iptv-org/community#1116] [Show and tell]
### Describe your issue

Channels are not playing in lists 
The channels are available in https://tvnpass.com/

### Contributing Guide

- [X] I have read [Contributing Guide](https://github.com/iptv-org/iptv/blob/master/CONTRIBUTING.md)

**Comment by KobayashiSan507** [comment]
TVMax Panama only broadcasts in YouTube  certain football matches and other national sports, not the rest of the programming such as movies, series and own programs

**Comment by R4IN2007** [comment]
New URLS? MDSTRM Ones...
TVN (Panama) - https://mdstrm.com/live-stream-playlist/64cbe53b46ee28089166c04f.m3u8?access_token=oFmiKkspUAiNwqCfnYlhCBAC7jsGwbfrlAZ8WDmh9mpMMlrS7K55BDY88IJHAWN23nwbCbWp5rE (URL Completo y sin access_token no sirve) Without access_token won't work

TVMax - https://mdstrm.com/live-stream-playlist/65d4c7bcbe741b6c4131bd70.m3u8?access_token=RW1Qup0gqJAZi0l3p81SkaF2KotrPO8E3ERj0VRuei9GHoIXRa9ag0hAJpBlT2iQLzRkBJ8KvHQ (URL Completo y sin access_token no sirve)
Without access_token won't work

Extras: TVN Radio 96.5FM [VIDEO]
https://mdstrm.com/live-stream-playlist/65d4c68470d6fd087fe7d8fb.m3u8?access_token=T1DcDYRs9VUySio4t7Y1qsuGIhIXVOjkUhM4ms4lXnGI5hpWsKb29PSp0132EqiRcdOx7cfpYxk - Without access_token won't work

Edit: Nervermind ; URL works just that the Access_token doesn't last long

![Screenshot_20240904_160849_Chrome](https://github.com/user-attachments/assets/c63aa88d-bde4-4c4d-8e6c-586f0da0fcf0)

**Comment by Toei79** [comment]
maybe getting a script to get token or that no possible?

**Comment by R4IN2007** [comment]
I guess but the token changes everytime when using the m3u8.

**Comment by R4IN2007** [comment]
Should do the same like for the channels in Issue iptv-org/iptv#17397

**Comment by Toei79** [comment]
> I guess but the token changes everytime when using the m3u8.

its possible do a  script that changes the stream link in the playlist , 
get the token link in a file then change to the playlist.

what i don't know how easy get the token stream to a file

**Comment by ghost** [comment]
> Should do the same like for the channels in Issue iptv-org/iptv#17397

i can make a script for that :)

**Comment by ghost** [comment]
hmm, it looks like the token in the api is being generated and iframed. thats how it works. but ima check later.

**Comment by ghost** [comment]
Okay, so i checked it out and it is google api. Basically you need a token to send a request for the essential info to run the stream. example: https://securetoken.googleapis.com/v1/token?key=KEY and you will receive something like this:

{
    "access_token": "nope",
    "expires_in": "3600",
    "token_type": "Bearer",
    "refresh_token": "nope",
    "id_token": "nope",
    "user_id": "nope",
    "project_id": "152696431546"
}

but you need only the access token to run the stream, so yeah. btw it is google api so idk. and i dont know where the key comes from for the google api.

**Comment by ghost** [comment]
import requests

headers = {
    'accept': '*/*',
    'accept-language': 'pl-PL,pl;q=0.9,en-US;q=0.8,en;q=0.7',
    'content-type': 'application/x-www-form-urlencoded',
    'origin': 'https://tvnpass.com',
    'priority': 'u=1, i',
    'referer': 'https://tvnpass.com/',
    'sec-ch-ua': '"Chromium";v="128", "Not;A=Brand";v="24", "Google Chrome";v="128"',
    'sec-ch-ua-mobile': '?0',
    'sec-ch-ua-platform': '"Windows"',
    'sec-fetch-dest': 'empty',
    'sec-fetch-mode': 'cors',
    'sec-fetch-site': 'cross-site',
    'user-agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/128.0.0.0 Safari/537.36',
    'x-client-data': 'maybe not',
    'x-client-version': 'Chrome/JsCore/7.14.1/FirebaseCore-web',
}

params = {
    'key': 'the key that is used for the api',
}

data = {
    'grant_type': 'refresh_token',
    'refresh_token': 'nope',
}

response = requests.post('https://securetoken.googleapis.com/v1/token', params=params, headers=headers, data=data)

**Comment by Toei79** [comment]
Ok thats python script?, it gets token numbers letters then replace token letters on the expired one?

I just think get toke stream in x.xxx file then use a replacing text script on the playlist  , running each  x time on crontab

**Comment by freearhey** [comment]
https://github.com/iptv-org/iptv/blob/5165b3b06d6391e19ae8134885593845d96e3918/streams/pa.m3u#L46-L49

**Reply by R4IN2007** [reply]
Nice, well now from media stream to brightcove
It better off it stays bcovlive then mdstrm

**Comment by R4IN2007** [comment]
Link changed again

**Comment by freearhey** [comment]
https://github.com/iptv-org/iptv/discussions/new?category=channel-search
